src/integrator/sycl/collocation_device.cpp

```diff
@@ -18,15 +18,23 @@ namespace sycl       {
                                  const T *pts_device, T *eval_device,
                                  cl::sycl::queue *queue) {
 
-        cl::sycl::range<2> threads(16, 16);
-        cl::sycl::range<2> blocks(util::div_ceil(npts, threads[0]),
-                                  util::div_ceil(nshells, threads[1]) );
+        cl::sycl::range<3> threads(16, 16, 1);
```

Comment:
Did you find out what the SYCL backend agnostic way of determining these things is? I'm in the process of removing "magic" numbers in the CUDA impl as well, would be good to get them sorted out here

Reply:
Done. Similar to `cuda`'s work